File: DotNet/Models/Reservation.cs
using System;
using System.Collections.Generic;

namespace Endpoints.Models;

public partial class Reservation
{
    public long Id { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public short GuestNumber { get; set; }

    public float TotalCost { get; set; }

    public long FlatId { get; set; }

    public long ReservedById { get; set; }

    public virtual Flat Flat { get; set; } = null!;

    public virtual User ReservedBy { get; set; } = null!;
}

File: Java/src/main/java/com/ormComp/model/Reservation.java
package com.ormComp.model;

import jakarta.persistence.*;

import java.time.OffsetDateTime;
import java.util.Objects;

@Entity
@Table(name = "reservations", schema = "rental")
public class Reservation {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id", nullable = false)
    private long id;
    @Column(name = "start_date", nullable = false)
    private OffsetDateTime startDate;
    @Column(name = "end_date", nullable = false)
    private OffsetDateTime endDate;
    @Column(name = "guest_number", nullable = false)
    private short guestNumber;
    @Column(name = "total_cost", nullable = false)
    private int totalCost;
    @ManyToOne
    @JoinColumn(name = "flat_id")
    private Flat flat;
    @ManyToOne
    @JoinColumn(name = "reserved_by_id")
    private User user;
}
